plugins/router/snmp__mikrotik

```diff
@@ -57,6 +58,7 @@ if (defined $ARGV[0] and $ARGV[0] eq "snmpconf") {
     print "require $sysFlashTotalOID\n";
     print "require $sysRAMUsageOID\n";
     print "require $sysRAMTotalOID\n";
+    print "require $sysTempOID\n";
```

Comment:
I'm not sure about this line as some routers didn't report temperature and I don't know if it could affect how the snmpconf parameter works for that routers.

Reply:
Probably you are right (I do not know SNMP plugins too well). Thus maybe just remove it and add a comment above, that only attributes available on all routers should be listed here as *required*.

At least this is my understanding.

Reply:
You are right, I found it in the file [SNMPConfig.pm](https://github.com/munin-monitoring/munin/blob/master/lib/Munin/Node/SNMPConfig.pm). All OIDs are checked and if any fails then  the plugins is not suggested, so the it should be removed. I'll do it.

But I found out that the requirements for the other OIDs are not working as expected. Just now the 'snmp__mikrotik' plugin is being recommended for any device, even if the OID is not present (at least in my system).
That happen because the `require {OID}` could have a second parameter like `require {OID} [{filter}]` and without a filter the plugin is suggested regardless of the OID. I've tested it against devices from other vendors (that they don't have any of the requested OIDs) and it get suggested anyway.

The lines should be something like:

`print "require $OID ^\\d\n";`

But I don't know if I should fix that in this PR or in another.

Reply:
The filter should be optional (at least from reading the code). But maybe the implementation of `_snmp_check_require` in munin is broken?

Just for testing a theory - maybe you could change the following line:
```perl
my $response = $session->get_request($oid);
```
into the following:
```perl
my $response = $session->get_request(-varbindlist => [ $oid ],);
```

At least this seems to be documented way for calling `get_request`. Does this fix the issue for you?

(I do not have an SNMP-capable device around, thus I cannot test it easily)

Reply:
I've tested adding the '-varbindlist' parameter and it has no effect. I've also tested if any error was returned when trying to get an inaccessible oid and no error was reported. I found this [bug](https://rt.cpan.org/Public/Bug/Display.html?id=131496) that it's not exactly the same but has a similar effect.
I noticed that other plugins like snmp__load or snmp__print_supplies just use the filter parameter.

Reply:
Thank you for trying it!
What a pity, that it did not work :(
I will try to debug this tomorrow at work - there are some SNMP switches around.

> I noticed that other plugins like snmp__load or snmp__print_supplies just use the filter parameter.

Anyway this would be a bug in munin (if I understand it correctly). Thus I would really like to fix it at its root instead of adding workarounds everywhere :)

Reply:
How do we proceed with this plugin now?

Reply:
I've been testing a perl script with only Net::SNMP and I found what looks like a bug in that perl library that didn't throw an error when trying to get an inaccessible oid when working with version 2c of snmp. I'm going to report it, maybe the cause is the same that the other bug.

We can wait until this is solved (but Net::SNMP doesn't looks very active) or I can upload the version I have with the filter parameter and merge it.

Reply:
Here is the [bur report](https://rt.cpan.org/Public/Bug/Display.html?id=132323) to Net::SNMP if you want to test it or try to found a solution.

Reply:
Thank you for digging further into this issue!

I would be glad, if you would take the following steps:
1. open an issue against munin describing the current problem with the `required` OID setting (I would have a hard time putting the details together, sorry)
2. apply your *filter* workaround to the plugin and reference the issue above (in order to remove the workaround somewhen in the future)

Thank you a lot for your attention to the details!

Reply:
Just done both things!